handler/src/main/java/io/netty/handler/ssl/SniCompletionEvent.java

```diff
@@ -25,16 +25,16 @@
 public final class SniCompletionEvent extends SslCompletionEvent {
     private final String hostname;
 
-    SniCompletionEvent(String hostname) {
+    public SniCompletionEvent(String hostname) {
         this.hostname = hostname;
```

Comment:
Shouldn't we add a `null` check? Because if SNI is present, there must be a value.

Reply:
Actually it might be null if we couldn't parse and select the default context. So reverting the null check

Reply:
In that case, we should throw exception directly instead of firing this event.

Since this is an event, we should have 2 event classes. One should be fired on successful completion and on should be fired on failure.

WDYT?

Reply:
no as for example no SNI extension might be present... Also I don't want to change how the even is used at the moment to not break users code